src/map.rs

```diff
@@ -995,3 +1027,21 @@ impl<'a, S> Streamer<'a> for StreamOutput<S>
         self.0.next().map(|(k, v)| (k, raw::Output::new(v)))
     }
 }
+
+/// A lexicographically ordered stream of key-value from a map
+/// along with the states of the automaton.
+///
+/// The `Stream` type is based on the `StreamWithState`.
+pub struct StreamWithState<'m, A=AlwaysMatch>(raw::StreamWithState<'m, A>) where A: Automaton;
+
+impl<'a, 'm, A: 'a + Automaton> Streamer<'a> for StreamWithState<'m, A>
+    where A::State: Clone
+{
+    type Item = (&'a [u8], u64, A::State);
+
+    fn next(&'a mut self) -> Option<Self::Item> {
+        /// My IDE Does not like this, and if I pub this function the compile fails, I'm a bit confused
```

Comment:
remove this comment.

Reply:
if I write `pub fn next(...)` instead of just `fn next(...)` the code does not compile